modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala

```diff
@@ -11,9 +16,31 @@ case object JavaPreprocessor extends Preprocessor {
     logger: Logger
   ): Option[Either[BuildException, Seq[PreprocessedSource]]] =
     input match {
-      case j: Inputs.JavaFile =>
-        Some(Right(Seq(PreprocessedSource.OnDisk(j.path, None, None, Nil, None))))
-
+      case j: Inputs.JavaFile => Some(either {
+          val content   = value(PreprocessingUtil.maybeRead(j.path))
+          val scopePath = ScopePath.fromPath(j.path)
+          val ExtractedDirectives(_, directives0) =
+            value(from(
```

Comment:
It would be nice if we could enrich the `DirectiveError` here with the information that it happened in the java context but the `DirectiveError` is an opaque class.